CDCgov/prime-reportstream
# 1960 Run CLI tests in parallel

This PR adds the ability for the PRIME CLI to run tests in parallel.  Understand that this uses co-routines and not threads, so tests will run as Kotlin determines resources are available without incurring into a lot of overhead, so you will see a few tests run and as they finish more will run.

To test:
1. Run the smoke test by running `./prime test` or `gradle testSmoke` and verify the tests run in parallel.
2. Verify the merge test is successful.
3. Run the smoke test in sequence by running `./prime test --sequential` and verify the test runs in sequence.
4. Run the repeat waters test by running `./prime test --run=repeatwaters` and verify the test runs.

## Changes
- Changed the merge test to rely on report IDs, so the test can run with others at the same time
- Tweaked the repeatWaters test to allow for its multi-threading
- Changes the wait function to use delay, which suspends the running of the coroutine
- Hold onto any messages to be printed by a test until the test is done, that way the output is organized
- Added an option to run the tests in sequence which prints the test output as it happens

## Checklist

### Testing

### Process

## Fixes
- #1960

## To Be Done
